packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -91,9 +91,21 @@ enum MainAxisSize {
 ///  * [RenderFlex], the flex render object.
 enum MainAxisAlignment {
   /// Place the children as close to the start of the main axis as possible.
+  ///
+  /// If this value is used in a horizontal direction, a [TextDirection] must be
+  /// available to determine if the start is the left or the right.
+  ///
+  /// If this value is used in a vertical direction, a [VerticalDirection] must be
+  /// available to determine if the start is the top or the bottom.
```

Comment:
It would be consistent with our philosophy to have start always be "top" if the main axis is vertical.